File: code/java/src/main/java/sdk/events/EventInterface.java
package sdk.events;

import sdk.enums.EventType;
import java.util.Map;

public interface EventInterface {

    /**
     * Returns the event type for this event.
     *
     * @return The event type.
     */
    EventType getEventType();

    /**
     * Returns the options for this event.
     *
     * @return A map of options.
     */
    Map<String, Object> getOptions();

    /**
     * Returns the data lines for this event.
     *
     * @return A list of strings representing the data lines.
     */
    String[] getDataLines();

    /**
     * Converts a boolean to its string representation.
     *
     * @param value The boolean value.
     * @return "true" if true, otherwise "false".
     */
    String getBooleanAsString(boolean value);

    /**
     * Constructs a data line from the given key and value.
     *
     * @param key   The key for the data line.
     * @param value The value for the data line.
     * @return A formatted data line string.
     */
    String getDataLine(String key, String value);
}

File: sdk/php/src/events/EventInterface.php
<?php
/**
 * @copyright Copyright (c) PutYourLightsOn
 */

namespace starfederation\datastar\events;

use starfederation\datastar\enums\EventType;

interface EventInterface
{
    /**
     * Returns the event type for this event.
     */
    public function getEventType(): EventType;

    /**
     * Returns the options for this event.
     *
     * @return string[]|int[]
     */
    public function getOptions(): array;

    /**
     * Returns the data lines for this event.
     *
     * @return string[]
     */
    public function getDataLines(): array;

    /**
     * Returns a boolean as a string.
     */
    public function getBooleanAsString(bool $value): string;

    /**
     * Returns a data line.
     */
    public function getDataLine(string $value): string;

    /**
     * Returns multiple data lines.
     */
    public function getMultiDataLines(string $literal, string $data): array;
}
